src/test/java/uk/gov/homeoffice/digital/sas/timecard/producers/KafkaProducerServiceTest.java

```diff
@@ -70,4 +82,106 @@ void sendMessage_actionOnResource_messageIsSentWithCorrectArguments(KafkaAction
     assertThat(messageArgument.getValue().getResource()).isEqualTo(timeEntry);
     assertThat(messageArgument.getValue().getAction()).isEqualTo(action);
   }
+
+  @ParameterizedTest
+  @EnumSource(value = KafkaAction.class, names = {"CREATE", "UPDATE", "DELETE"})
+  void CompletableFutureReporting_actionOnResource_onFailureMessageLogged(KafkaAction action) throws ExecutionException, InterruptedException {
+    ReflectionTestUtils.setField(kafkaProducerService, "topicName", TOPIC_NAME);
+    ReflectionTestUtils.setField(kafkaProducerService, "projectVersion", "1.0.0");
```

Comment:
nit: We could probs move this and the occurrence in the below tests to a `@BeforeEach` method since we always need it for the service to work

Reply:
Actioned